### ISSUE TITLE
Expensive rendering could defer tab first rendering

### ISSUE THREAD

**Issue by wangbourne** [react-native-community/react-native-tab-view#493]
### Current behaviour
I have around 30 tabs,  it's OK if my `renderScene ` is not an expensive rendering. However, if it's an expensive rendering (For example long list etc), there always come a delay to for first rendering of selected tab.
By the way, I have made use of https://github.com/react-native-community/react-native-tab-view#optimization-tips, including using FlatList. Expensive scene rendering seems unavoidable in my application :)

### Expected behaviour
Selected tab (including indicator) should come without being blocked by other rendering, which is more natural and reasonable.
Is there a trick to show selected tab before the tab scene is rendered, so we can avoid above issue?

### Screenshots (if applicable)
![demo](https://user-images.githubusercontent.com/2090084/37343088-2cdce516-2712-11e8-9d5c-ce86380615b1.gif)

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        |   iPhone5s with iOS8, 10 and 11
| react-native          |  0.51.10
| react-native-tab-view |  0.0.75
| node                  |    v9.4.0
| npm or yarn           |    1.5.1 (Yarn)

**Comment by wangbourne** [comment]
Potential fix could be set a correct `initialOffset` in constructor of tabbar.js since `this.props.layout.width` returns 0 in constructor.
```
//tabbar.js
const initialOffset =
      this.props.scrollEnabled && this.props.layout.width
        ? {
            x: this._getScrollAmount(
              this.props,
              this.props.navigationState.index
            ),
            y: 0,
          }
        : undefined;

```

**Comment by satya164** [comment]
Sounds reasonable. Can you send a PR?

**Comment by zabojad** [comment]
Version 1.3.0 should improve your use case thanks to https://github.com/react-native-community/react-native-tab-view/commit/5b0a661b24ea587a0e4c282c6cbbd553d402877e

**Comment by satya164** [comment]
Hey, I just released a new alpha `2.0.0-alpha.0` of the library. It's rewritten using [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) and [react-native-reanimated](https://github.com/kmagiera/react-native-reanimated) addresses a many platform specific bugs and performance problems. The documentation is updated as well.

Please try the new version and see if it addresses your issue. If not, please open a new issue following the issue template.
